bin/generate-php-sync-issue.mjs

```diff
@@ -0,0 +1,460 @@
+
+		for ( const [ key, value ] of Object.entries( levelData ) ) {
+			if ( Array.isArray( value ) ) {
+				if ( value.length === 1 && parentData && parentKey ) {
+					if ( ! Array.isArray( parentData[ parentKey ] ) ) {
+						parentData[ parentKey ] = [];
+					}
+					parentData[ parentKey ] = [
+						...parentData[ parentKey ],
+						...value,
+					];
+				} else {
+					processedData[ key ] = value;
+				}
+			} else {
+				processedData[ key ] = processLevel(
+					value,
+					processedData,
+					key
+				);
+			}
+		}
+
+		return processedData;
+	}
+
+	return processLevel( data );
+}
+
+function removeNesting( data, maxLevel = 3 ) {
+	function processLevel( levelData, level = 1 ) {
+		const processedData = {};
+
+		for ( const [ key, value ] of Object.entries( levelData ) ) {
+			if ( Array.isArray( value ) ) {
+				processedData[ key ] = value;
+			} else if ( level < maxLevel ) {
+				processedData[ key ] = processLevel( value, level + 1 );
+			} else {
+				processedData[ key ] = flattenData( value );
+			}
+		}
+
+		return processedData;
+	}
+
+	function flattenData( nestedData ) {
+		let flatData = [];
+
+		for ( const value of Object.values( nestedData ) ) {
+			if ( Array.isArray( value ) ) {
+				flatData = [ ...flatData, ...value ];
+			} else {
+				flatData = [ ...flatData, ...flattenData( value ) ];
+			}
+		}
+
+		return flatData;
+	}
+
+	return processLevel( data );
+}
+
+function processCommits( commits ) {
+	const result = {};
+
+	commits.forEach( ( commit ) => {
+		// Skip commits without an associated pull request
+		if ( ! commit.pullRequest ) {
+			return;
+		}
+		commit.files.forEach( ( file ) => {
+			// Skip files that are not PHP files.
+			if ( ! file.filename.endsWith( '.php' ) ) {
+				return;
+			}
+
+			// Skip files within specific packages.
+			if (
+				IGNORED_PATHS.some(
+					( path ) =>
+						file.filename.startsWith( path ) ||
+						file.filename === path
+				)
+			) {
+				return;
+			}
+
+			const parts = file.filename.split( '/' );
+			let current = result;
+
+			// If the file is under 'phpunit', add it directly to the 'phpunit' key
+			// this is it's helpful to have a full list of commits that modify tests.
+			if ( parts.includes( 'phpunit' ) ) {
+				current.phpunit = current.phpunit || [];
+				current.phpunit = [ ...current.phpunit, commit ];
+				return;
+			}
+
+			for ( let i = 0; i < parts.length; i++ ) {
+				const part = parts[ i ];
+
+				// Skip 'src' part under 'block-library'
+				if ( part === 'src' && parts[ i - 1 ] === 'block-library' ) {
+					continue;
+				}
+
+				if ( i === parts.length - 1 ) {
+					current[ part ] = current[ part ] || [];
+					current[ part ] = [ ...current[ part ], commit ];
+				} else {
+					current[ part ] = current[ part ] || {};
+					current = current[ part ];
+				}
+			}
+		} );
+	} );
+
+	return result;
+}
+
+function formatPRLine( pr ) {
+	return `- [ ] ${ pr.url } - @/${ pr.creator } | Trac ticket | Core backport PR\n`;
```

Comment:
```suggestion
	return `- [ ] ${ pr.url } - @${ pr.creator } | Trac ticket | Core backport PR\n`;
```

We'll need to commit this before we merge this PR.

Reply:
Should be fine to just commit it? Nobody will actually be pinged until the issue is created, right?